[KEYCLOAK-10694] Rebase the Red Hat Single Sign-On container image on top of the Red Hat Universal Base Image 8 Minimal image. on top of the Red Hat Universal Base Image 8 Minimal image. [KEYCLOAK-10124] Define RHEL-8 content set

iankko opened the pull request:
[KEYCLOAK-10694] Rebase the Red Hat Single Sign-On container image
    on top of the Red Hat Universal Base Image 8 Minimal image.
    Default to OpenJDK JDK 11
    
    Also:
    * Update the CCT module branch to Sprint #35,
    * Update jboss-eap-7-image tag to CD16,
    * **Default to OpenJDK JDK 11**,
    * Introduce 'overrides' subdirectory as a common location to
      keep examples of various CEKit Overrides files. Provide
      example on how to utilize the Overrides file to include JDK 8
      into the image using both CEKit 2.x and CEKit 3.x versions,
    * Introduce 'backward-compatibility' subdirectory as a common
      location to keep various CEKit image descriptor files, that
      need to be archived each time a major, backward-incompatible
      change is introduced to CEKit. Archive image file descriptor,
      used to base the image on RHEL-7, containing 'jboss-os'
      repository definition, and using yum to install the RPM packages.
    
    Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>
--

    [KEYCLOAK-10124] Define RHEL-8 content set (with RHEL-8 UBI BaseOS
    repo definition) for OSBS builds
    
    Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>



Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for other issues
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

Commit:
[KEYCLOAK-10694] Rebase the Red Hat Single Sign-On container image
on top of the Red Hat Universal Base Image 8 Minimal image.
Default to OpenJDK JDK 11

Also:
* Update the CCT module branch to Sprint #35,
* Update jboss-eap-7-image tag to CD16,
* **Default to OpenJDK JDK 11**,
* Introduce 'overrides' subdirectory as a common location to
  keep examples of various CEKit Overrides files. Provide
  example on how to utilize the Overrides file to include JDK 8
  into the image using both CEKit 2.x and CEKit 3.x versions,
* Introduce 'backward-compatibility' subdirectory as a common
  location to keep various CEKit image descriptor files, that
  need to be archived each time a major, backward-incompatible
  change is introduced to CEKit. Archive image file descriptor,
  used to base the image on RHEL-7, containing 'jboss-os'
  repository definition, and using yum to install the RPM packages.

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

iankko commented:
@hmlnarik @slaskawi @douglaspalmer @martin-kanis PTAL once got a change

Thanks! Jan

slaskawi reviewed (approved):
LGTM! Great work!

Could you please paste somewhere what's the new image size? I wonder how much MB did we shave off with this PR.

Also, could you please paste a diff of installed packages? I guess this should shrink as well.

hmlnarik reviewed (commented):
Looks good to me, thanks! Not approving/merging though, since this is still marked as [WIP]

iankko commented:
> Looks good to me, thanks! Not approving/merging though, since this is still marked as [WIP]

@hmlnarik Right, good catch. Was solving something yesterday WRT to this yet & forgot to remove it. But in any case, it's OK this not to be merged immediately TBH (rather than only in moment when also corresponding change for the OpenShift image is ready for review / merge). The reason is I am not sure, some changes for this one won't be needed due to this one yet. so better to have them merged all or none style. Thanks!

iankko commented:
@slaskawi Updated the version from yesterday a bit yet. Due to [RH BZ#1725863](https://bugzilla.redhat.com/show_bug.cgi?id=1725863) needed to stick the RHEL-8 UBI Minimal base image to some fixed version, rather than using `latest` till the moment that RH BZ is corrected. jFYI.

Regarding the stats yet. Dont' have all of them (the OpenShift image still needs to be done). But for the standalone one, I would say the size reduction is pretty significant.

Here are the stats for the standalone image, where RHEL-8 UBI minimal is the base image:
```
Original image size: 1387.66 MB
Squashed image size: 664.99 MB
```
while for RHEL-7 based one it was (picked up some random recent previous build):
```
Original image size: 1368.41 MB
Squashed image size: 1290.13 MB
```

So it's more than a 1/2 size of the previous case I would say. The stats for OpenShift image to arrive yet (once I have them).

slaskawi commented:
@iankko Thanks! That looks super-promising!

iankko commented:
@hmlnarik @slaskawi Refactored to the original to preserve backward-compatibility (reproducible image (re)-builds with CEKit 2.x versions for RH-SSO CD image versions 6.0 and earlier).

This should be ready for review again.

Commit:
[KEYCLOAK-10124] Define RHEL-8 content set (with RHEL-8
UBI BaseOS repo definition) for OSBS builds

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

Commit:
[KEYCLOAK-10694] Update OSBS image name, component name, and branch names
(both stable and -dev ones) for RH-SSO image based on the RHEL-8 UBI
minimal image

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

iankko commented:
OSBS information (branch, build target & component name) updated now too. This is ready for review.

hmlnarik reviewed (approved):
Thanks @iankko !
